core/trino-main/src/main/java/io/trino/sql/planner/plan/GroupIdNode.java

```diff
@@ -129,20 +135,19 @@ public <R, C> R accept(PlanVisitor<R, C> visitor, C context)
 
     public Set<Symbol> getInputSymbols()
     {
-        return ImmutableSet.<Symbol>builder()
-                .addAll(aggregationArguments)
-                .addAll(groupingSets.stream()
-                        .map(set -> set.stream()
-                                .map(groupingColumns::get).collect(Collectors.toList()))
-                        .flatMap(Collection::stream)
-                        .collect(toSet()))
```

Comment:
I think just switching this to `toImmutableSet()` would do it.

Reply:
It would, but this version avoids materializing an intermediate `List` for each grouping set, pre-sizes the builder, and avoids a bunch of extra lambda creations per call. That said, I'm happy to switch it back to the less efficient version (but with an `ImmutableSet`) if you prefer.

Reply:
I'm ok either way, but *that* would deserve a separate change for the purpose of performance improvement (and only if we see that it matters)

Reply:
Ok, I've split the stability fix of using `toImmutableSet()` from the method cleanup into separate commits.